src/tabbar.cpp

```diff
@@ -463,7 +463,7 @@ void TabBar::paintEvent(QPaintEvent *e)
 
         int align = Qt::AlignVCenter;
         if (need_fade_out) {
-            align |= (right_to_left)? Qt::AlignRight : Qt::AlignLeft;
+            align |= (textRightToLeft == appRightToLeft) ? Qt::AlignLeft : Qt::AlignRight;
```

Comment:
This looks strange. Why is the case of `textRightToLeft` and `appRightToLeft` both being `false` aligned the same way as when both of them are `true`?

Reply:
I will retest this and either make a explanation and visual, or correct it. A comment I believe will be needed here to explain the textbox behaviours.

Reply:
The alignment in RTL is flipped, i.e. if you set Align Left, it will actually align it to the right. https://doc.qt.io/qt-6/qtquick-positioning-righttoleft.html.